### ISSUE TITLE
bug: required fields are not enforced in the modal

### ISSUE THREAD

**Issue by kymppi** [formbricks/formbricks#230]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

I noticed that you could hit the finish button on the modal, it would show that fields are required but it still submitted the answer to formbricks.

### Steps to Reproduce

1. Use the default PMF survey
2. Click next on the modal in next.js app
3. Click finish on the modal
4. See that a field is required
5. Still see the answer in formbrick's UI

## Environment

- [x] Formbricks Cloud (app.formbricks.com)
- [ ] self-hosted snoopForms, version/commit: [please provide]

### Additional Context

(Write your answer here.)

**Comment by ramsaysewell** [comment]
I'd like to have a go at this, I'll see how far I get and will keep you updated if I come across any issues with reproducing or indeed the fix.

**Comment by ramsaysewell** [comment]
After spending quite a bit of time looking through the process and debugging the user flow, I've realised that the responseId is created as soon as the user starts clicking through the survey. Therefore, the only logic stopping a response being created in Formbrick's UI is the checking of the questions value (to then proceed to the next one). 

Currently, there is no validation in the Multiple Choice type questions so setting a question of that type to "required" does not stop the user from skipping this question.

I've tested this theory and creating a survey with 1 required open text question, the response does not show in the UI until text is present and the user hits "next" or "finish" - this is basic low level html `required` attribute on the `<textarea />`.

I think going forward, the multiple choice question will need validation added to it and either we continue with the behaviour of the response coming through as a stream or we change the behaviour to only submit/create a response when the user hits "finish".

Happy to hear thoughts on this and more than happy to answer any questions.

Nevertheless, an exciting PR is in our midst!
